File: sdk/typescript/examples/basic.ts
/**
 * Basic example using the RMK TypeScript SDK
 */

import { RMKClient } from "../src/client.js";

async function main() {
  // Initialize client
  const client = new RMKClient({
    baseURL: "http://localhost:9090",
    timeout: 30000,
  });

  try {
    // Login
    console.log("Logging in...");
    const auth = await client.login("user", "password");
    console.log("Logged in as:", auth.username);

    // Store a memory
    console.log("\n--- Storing memory ---");
    const memory = await client.memoryStore({
      namespace: "user_123",
      content: "Claude Desktop is an AI assistant that can use MCP tools",
      nodeType: "Fact",
      tags: ["ai", "mcp", "claude"],
    });
    console.log("Stored memory:", memory);

    // Search memories
    console.log("\n--- Searching memories ---");
    const results = await client.memorySearch({
      namespace: "user_123",
      query: "Claude Desktop",
      limit: 5,
    });
    console.log(`Found ${results.count} memories:`, results.results);

    // Chat consultation
    console.log("\n--- Chat consultation ---");
    const chat = await client.chatConsult({
      namespace: "user_123",
      message: "What do you know about Claude Desktop?",
    });
    console.log("Response:", chat.response);

    // Create an entity
    console.log("\n--- Creating entity ---");
    const entity = await client.entityCreate({
      namespace: "user_123",
      name: "Claude",
      entityType: "Person",
      description: "AI assistant by Anthropic",
      relationships: [
        { type: "WORKS_ON", target: "anthropic" },
      ],
    });
    console.log("Created entity:", entity);

    // List available tools
    console.log("\n--- Available tools ---");
    const tools = await client.toolsList();
    console.log(`Total tools: ${tools.tools.length}`);
    tools.tools.forEach((tool) => {
      console.log(`  - ${tool.name}: ${tool.description}`);
    });

  } catch (error) {
    console.error("Error:", error);
  }
}

main();

File: sdk/python/examples/basic.py
"""
Basic example using the RMK Python SDK
"""

import asyncio
from rmk import RMKClient, NodeType, RelationshipType


def main():
    """Synchronous example."""
    with RMKClient(base_url="http://localhost:9090") as client:
        # Login
        print("Logging in...")
        auth = client.login("user", "password")
        print(f"Logged in as: {auth.username} (role: {auth.role})")

        # Store a memory
        print("\n--- Storing memory ---")
        memory = client.memory_store(
            namespace="user_123",
            content="Claude Desktop is an AI assistant that can use MCP tools",
            node_type=NodeType.FACT,
            tags=["ai", "mcp", "claude"],
        )
        print(f"Stored memory: {memory.uid}")

        # Search memories
        print("\n--- Searching memories ---")
        results = client.memory_search(
            namespace="user_123",
            query="Claude Desktop",
            limit=5,
        )
        print(f"Found {results.count} memories")
        for node in results.results:
            print(f"  - {node.get('name', 'N/A')}: {node.get('description', 'N/A')}")

        # Chat consultation
        print("\n--- Chat consultation ---")
        chat = client.chat_consult(
            namespace="user_123",
            message="What do you know about Claude Desktop?",
        )
        print(f"Response: {chat.response}")

        # Create an entity
        print("\n--- Creating entity ---")
        entity = client.entity_create(
            namespace="user_123",
            name="Claude",
            entity_type="Person",
            description="AI assistant by Anthropic",
            relationships=[RelationshipType.WORKS_ON],
        )
        print(f"Created entity: {entity.uid}")

        # List available tools
        print("\n--- Available tools ---")
        tools = client.tools_list()
        print(f"Total tools: {len(tools.tools)}")
        for tool in tools.tools:
            print(f"  - {tool.get('name', 'N/A')}: {tool.get('description', 'N/A')}")


async def async_main():
    """Asynchronous example."""
    async with RMKClient(base_url="http://localhost:9090") as client:
        # Login
        print("Logging in...")
        auth = await client.alogin("user", "password")
        print(f"Logged in as: {auth.username} (role: {auth.role})")

        # Store a memory
        print("\n--- Storing memory ---")
        memory = await client.amemory_store(
            namespace="user_123",
            content="Claude Desktop is an AI assistant that can use MCP tools",
            node_type=NodeType.FACT,
            tags=["ai", "mcp", "claude"],
        )
        print(f"Stored memory: {memory.uid}")

        # Search memories
        print("\n--- Searching memories ---")
        results = await client.amemory_search(
            namespace="user_123",
            query="Claude Desktop",
            limit=5,
        )
        print(f"Found {results.count} memories")


if __name__ == "__main__":
    # Run sync example
    main()

    # Or run async example
    # asyncio.run(async_main())
